### PR TITLE
Remove unnecessary static parameters

### PR DESCRIPTION
**Pull request by devmotion**
Removes some static parameters that are not needed (https://docs.julialang.org/en/v1/manual/style-guide/#Don't-use-unnecessary-static-parameters).

### CODE REVIEW

File: src/bracketing.jl [hunk 1]
```diff
@@ -262,7 +259,7 @@ function find_zero(fs, x0, method::Bisection;
                    p=nothing,
                    tracks = NullTracks(),
                    verbose=false,
-                   kwargs...) where {M <: Union{Bisection}}
+                   kwargs...)
```
Comment (devmotion) [inline src/bracketing.jl:262]: I guess at some point the function signature was `method::M, ...` and it was forgotten to remove the where clause when it was changed.

File: src/find_zero.jl [hunk 2]
```diff
@@ -271,90 +271,84 @@ end
...
-(F::Callable_Function{S,T,𝑭,P})(x) where {𝐍, S <: Val{𝐍}, T <: Val{true}, 𝑭, P<:Nothing} = begin
+function (F::Callable_Function{Val{𝐍},Val{true},𝑭,Nothing})(x) where {𝐍, 𝑭}
     fs = ntuple(i -> F.f[i](x), Val(𝐍))
     first(fs), ntuple(i -> fs[i]/fs[i+1], Val(𝐍-1))
 end
 
-(F::Callable_Function{S,T,𝑭,P})(x) where {𝐍, S <: Val{𝐍}, T <: Val{true}, 𝑭, P} = begin
-    fs = ntuple(i -> F.f[i](x, p), Val(𝐍))
+function (F::Callable_Function{Val{𝐍},Val{true},𝑭,P})(x) where {𝐍, 𝑭, P}
+    fs = ntuple(i -> F.f[i](x, F.p), Val(𝐍))
```
Comment (devmotion) [inline src/find_zero.jl:351]: This was a bug that was revealed by the language server.

File: src/bracketing.jl (hunk 1 above)
Comment (jverzani) [reply src/bracketing.jl:262]: Hmm, this one may be hiding something. There is a different criteria for exact bisection. Let me look.

File: src/find_zero.jl (hunk 2 above)
Comment (jverzani) [reply src/find_zero.jl:351]: Wow, better living through better tools...

**Comment by jverzani** [comment]
This is great! Thanks for submitting this! Let me double check the one comment, and merge in. Want to bump the version number so we can tag it?

File: src/bracketing.jl (hunk 1 above)
Comment (jverzani) [reply src/bracketing.jl:262]: Yup, you guessed right

**Comment by jverzani** [comment]
I'll merge and bump the version. Thanks again!

**Pull request merged**